[Vavinan] iP

Vavinan opened the pull request.

Vavinan committed:
Rename chatbot Duke to Zoro

Vavinan committed:
Level 0: Add Greet and Exit message

Vavinan committed:
Add Echo Feature

Vavinan committed:
Add Task List Feature

Vavinan committed:
Add Task Class

Vavinan committed:
Add Task Management Feature

Vavinan committed:
Format code to follow Coding Standards

Vavinan committed:
Add TaskManager and UserInterface classes

Vavinan committed:
Add Deadline, Event, and Todo classes.

Vavinan committed:
Inherit Deadline, Event, and Todo from Task class.

Vavinan committed:
Add Support for Tracking three types of Tasks.

wjunjie01 commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,60 @@
+
+    public void addTask(String taskDescription) {
+        if (taskDescription.startsWith("deadline")) {
+            String[] parts = taskDescription.substring(8).split("/by");
```
Magic number '8' can afford to make it final BEGIN_DEADLINE_INDEX

wjunjie01 commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,60 @@
+                return;
+            }
+        } else if (taskDescription.startsWith("event")) {
+            String[] parts = taskDescription.substring(5).split("/from|/to");
```
Magic number

wjunjie01 commented on src/main/java/Zoro.java:
```diff
@@ -0,0 +1,33 @@
+                taskManager.printTaskList();
+
+            } else if (input.startsWith("mark ")) {
+                taskManager.markTask(Integer.parseInt(input.substring(5).trim()) - 1);
```
Magic number

wjunjie01 commented on src/main/java/Zoro.java:
```diff
@@ -0,0 +1,33 @@
+                taskManager.markTask(Integer.parseInt(input.substring(5).trim()) - 1);
+
+            } else if (input.startsWith("unmark ")) {
+                taskManager.unmarkTask(Integer.parseInt(input.substring(7).trim()) - 1);
```
magic number

wjunjie01 reviewed (commented):
Overall coding standard looks quite okay, minor issue is the magic number. Good job!

Vavinan committed:
Update: if-else to switch, refine constants

Vavinan committed:
Add exceptions: Inavlid Task Format

Vavinan committed:
Extend Exception for Deadline, Event, and Todo Formats

Vavinan committed:
Implement Exception Handling for Task Operations, Mark, Unmark

Vavinan committed:
Merge branch 'branch-Level-5'

Vavinan committed:
Refactor TaskManager and UserInterface to use Java Collections

Vavinan committed:
Add support for deleting tasks from the list

Vavinan committed:
Feature: Implement toFileString() in Task and its subclasses

Vavinan committed:
Implement local task save feature

Vavinan committed:
Create tasks.txt and add initial tasks

Vavinan committed:
Merge branch 'branch-Level-6'

Vavinan committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/TaskManager.java

wwweert123 commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,215 @@
+    public static final int PART_1 = 1;
+    public static final int PART_2 = 2;
+    public static final int START_INDEX = 0;
+    private ArrayList<Task> taskList = new ArrayList<>();
```
the coding standard would prefer that you use plural form for collection variable. So for instance "tasks" would be better name

wwweert123 commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,215 @@
+        String taskType = taskDescription.split(" ")[START_INDEX];
+        try {
+            switch (taskType) {
+                case "deadline":
```
case here would have the same indentation level as switch according to coding standards

wwweert123 commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,215 @@
+    }
+
+    private void addDeadlineTask(String taskDescription) throws InvalidDeadlineFormatException {
+        String[] taskDetails = taskDescription.substring(DEADLINE_BEGIN_INDEX).split("/by");
```
Can declare the "/by" constant at the top as well to prevent the use of magic literals

wwweert123 commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,215 @@
+        }
+    }
+
+    private void addDeadlineTask(String taskDescription) throws InvalidDeadlineFormatException {
```
Can make use of more OOP by using a parser to parse inputs and command subclass such as "AddCommand" to execute the adding of the task to the task list. Can take reference from personsbook for implementation

wwweert123 commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,215 @@
+    private Task parseTaskFromString(String line) {
+        Task task = null;
+        String[] parts = line.split(" \\| ");
+        if (parts.length >= 3) {
```
Avoid magic numbers

wwweert123 reviewed (commented):
Overall, neat and organized code with good coding standards and high code quality, can implement more OOP in the future taking reference from personsbook

wwweert123 commented on src/main/java/UserInterface.java:
```diff
@@ -0,0 +1,123 @@
+import Exceptions.InvalidDeadlineFormatException;
+import Exceptions.InvalidEventFormatException;
```
packages name should be in all lowercase

Vavinan committed:
Add new Exception classes: LoadFileException and SaveFileException

Vavinan committed:
Add: Storage, Parser, and CommandHandler

This commit introduces significant changes to the Zoro application:
- Add Storage class to handle loading and saving task data from/to a file.
- Introduce Parser class to parse task strings into Task objects
- Add CommandHandler class to process user commands
- Modify Zoro class to integrate CommandHandler
- Update TaskManager, UserInterface classes to support changes

Vavinan committed:
Move classes to new package

Moved the following classes to a new package:
- Storage
- Parser
- UserInterface
- TaskManager
- CommandHandler
- Task
- Todo
- Event
- Deadline

This relocation organizes the codebase better and improves maintainability.

Vavinan committed:
Refactor: Update string constants

Vavinan committed:
Add: InvalidDateTimeFormatException class

Added a new exception class, InvalidDateTimeFormatException, to handle invalid date and time format errors.

Vavinan committed:
Refactor: Use constants in Parser class

Vavinan committed:
Refactor: Update classes to use Date and Time format

- Update CommandHandler, UserInterface, TaskManager, Event,
  and Deadline classes to support date and time format for
  deadline and event tasks.

- This refactor ensures that the application can handle date and
  time information correctly when parsing and processing task
  details.

Vavinan committed:
Implement Find method to search for tasks by keyword

Vavinan committed:
Enhance readability with comprehensive documentation

Vavinan committed:
Merge pull request #1 from Vavinan/branch-Level-8

Add branch-level-8 feature: Date and Time support for tasks.

Vavinan committed:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	src/main/java/Helper/UserInterface.java

Vavinan committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/Helper/Deadline.java
#	src/main/java/Helper/Event.java
#	src/main/java/Helper/TaskManager.java
#	src/main/java/Helper/UserInterface.java

Vavinan committed:
Merge pull request #2 from Vavinan/branch-Level-9

Add: Find tasks by Keyword from branch-level-9

Vavinan committed:
Merge branch 'master' into branch-A-JavaDoc

Vavinan committed:
Merge pull request #3 from Vavinan/branch-A-JavaDoc

Integrate Documentation from branch-A-JavaDoc

Vavinan committed:
Enhance console output formatting for readability

Vavinan committed:
Enhance: Implement Folder Creation if Nonexistent

Vavinan committed:
Update README.md

-Update README.md to display User Guide consist of
   features, installation guide, and usage.

Vavinan committed:
Update README.md

Vavinan committed:
Update README.md

-Update README.md to display User Guide consist of
   features, installation guide, and usage.

Vavinan committed:
Update README.md: Commands and User tips

This update consists of
  - User Commands section for task formats and examples
  - Useful tips section for better user guidance

Vavinan committed:
Refactor: Update access modifiers in UserInterface and TaskManager

 - Changed all methods in UserInterface to public static void for consistency and utility.
 - Converted specific TaskManager functions from public to protected for better encapsulation.
 - Made corresponding changes in other files to support the access modifier updates.

Vavinan committed:
Refactor: Fix warnings, add exception handling

 - Update code to address compiler warnings and ensure code quality.
 - Implement exception handling for keyword format errors in CommandHandler.

Vavinan committed:
Merge branch 'master' of https://github.com/Vavinan/ip

Vavinan committed:
Refactor: Remove excess constructor docs from exceptions.

Vavinan committed:
Update docs: UserInterface and CommandHandler

Vavinan committed:
Refactor: Move classes into package

 - Move Todo, Event, and Deadline into TaskType package
 - Update TaskManager and Parser to support changes

Vavinan committed:
Update: Set manifest version to 2.0

Vavinan committed:
Refactor packages name

Vavinan committed:
Update README.md

Vavinan committed:
Reformat Markdown syntax

Vavinan committed:
Reformat Markdown syntax in README.md

Vavinan committed:
Update Installation steps

Vavinan committed:
Refactor Command tips in README.md

Vavinan committed:
Reformat Command tips table structure
